From patchwork Thu Mar 30 19:29:54 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,3/3] config: am335x_evm: detect BeagleBone Blue using BLA
From: robertcnelson@gmail.com
X-Patchwork-Id: 745391
Message-Id: <20170330192954.22673-3-robertcnelson@gmail.com>
To: u-boot@lists.denx.de
Cc: Tom Rini <trini@konsulko.com>, Jason Kridner <jkridner@beagleboard.org>, 
 Will Newton <willn@resin.io>
Date: Thu, 30 Mar 2017 14:29:54 -0500

BeagleBone Blue is next grenation of boards from BeagleBoard.org, focusing
on robotics with a TI wl1835 wireless module for connectivity.

This board can be indentified by the BLAx value after A335BNLT (BBB)
in the at24 eeprom:
BLAx: [aa 55 33 ee 41 33 33 35  42 4e 4c 54 42 4c 41 30 |.U3.A335BNLTBLA2|]

http://beagleboard.org/blue
https://github.com/beagleboard/beaglebone-blue

firmware: https://github.com/beagleboard/beaglebone-black-wireless/tree/master/firmware
wl18xx mac address: /proc/device-tree/ocp/ethernet@4a100000/slave@4a100200/mac-address

Upstream-Status: Pending

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
CC: Tom Rini <trini@konsulko.com>
CC: Jason Kridner <jkridner@beagleboard.org>
CC: Will Newton <willn@resin.io>
---
 board/ti/am335x/board.c      | 4 ++++
 include/configs/am335x_evm.h | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/board/ti/am335x/board.c b/board/ti/am335x/board.c
index 54589638b3..3e842d3187 100644
--- a/board/ti/am335x/board.c
+++ b/board/ti/am335x/board.c
@@ -660,6 +660,10 @@ int board_late_init(void)
 		if (!strncmp(board_ti_get_rev(), "GW1", 3)) {
 			name = "BBGW";
 		}
+		/* BeagleBoard.org BeagleBone Blue */
+		if (!strncmp(board_ti_get_rev(), "BLA", 3)) {
+			name = "BBBL";
+		}
 	}
 
 	if (board_is_bbg1())
diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index d90c5156d3..5fbc1e3ff9 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -144,6 +144,8 @@
 			"setenv fdtfile am335x-bonegreen.dtb; fi; " \
 		"if test $board_name = BBGW; then " \
 			"setenv fdtfile am335x-bonegreen-wireless.dtb; fi; " \
+		"if test $board_name = BBBL; then " \
+			"setenv fdtfile am335x-boneblue.dtb; fi; " \
 		"if test $board_name = A33515BB; then " \
 			"setenv fdtfile am335x-evm.dtb; fi; " \
 		"if test $board_name = A335X_SK; then " \
